bench/bench.ml

```diff
@@ -0,0 +1,480 @@
+        benchmark = write_rev_seq_hash;
+        dependency = None;
+      };
+      {
+        name = "iter_rw";
+        synopsis = "[RW] Iter";
+        benchmark = iter;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_ro";
+        synopsis = "[RO] Find in random order";
+        benchmark = find_random_ro;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_random_rw";
+        synopsis = "[RW] Find in random order";
+        benchmark = find_random;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_ro";
+        synopsis = "[RO] Find absent values";
+        benchmark = find_absent_ro;
+        dependency = Some "replace_random";
+      };
+      {
+        name = "find_absent_rw";
+        synopsis = "[RW] Find absent values";
+        benchmark = find_absent;
+        dependency = Some "replace_random";
+      };
+    ]
+end
+
+let list_benchs () =
+  let pp_bench ppf b = Fmt.pf ppf "%s\t-- %s" b.Index.name b.synopsis in
+  Index.suite |> Fmt.(pr "%a" (list ~sep:Fmt.(const string "\n") pp_bench))
+
+let schedule p s =
+  let todos = List.map fst in
+  let init = ref (s |> List.map (fun b -> (p b.Index.name, b))) in
+  let apply_dep s =
+    let deps =
+      s
+      |> List.fold_left
+           (fun acc (todo, b) ->
+             if todo then
+               match b.Index.dependency with Some s -> s :: acc | None -> acc
+             else acc)
+           []
+    in
+    s |> List.map (fun (todo, b) -> (todo || List.mem b.Index.name deps, b))
+  in
+  let next = ref (apply_dep !init) in
+  while todos !init <> todos !next do
+    init := !next;
+    next := apply_dep !init
+  done;
+  let r = List.filter fst !init |> List.map snd in
+  r
+
+type config = {
+  key_size : int;
+  value_size : int;
+  nb_entries : int;
+  log_size : int;
+  seed : int;
+  with_metrics : bool;
+}
+[@@deriving yojson]
+
+let pp_config fmt config =
+  Format.fprintf fmt
+    "Key size: %d@\n\
+     Value size: %d@\n\
+     Number of bindings: %d@\n\
+     Log size: %d@\n\
+     Seed: %d@\n\
+     Metrics: %b" config.key_size config.value_size config.nb_entries
+    config.log_size config.seed config.with_metrics
+
+let cleanup root =
+  let files = [ "data"; "log"; "lock"; "log_async"; "merge" ] in
+  List.iter
+    (fun (b : Index.suite_elt) ->
+      let dir = root // b.name // "index" in
+      List.iter
+        (fun file ->
+          let file = dir // file in
+          if Sys.file_exists file then Unix.unlink file)
+        files)
+    Index.suite
+
+let init config =
+  Printexc.record_backtrace true;
+  Random.init config.seed;
+  if config.with_metrics then (
+    Metrics.enable_all ();
+    Metrics_gnuplot.set_reporter ();
+    Metrics_unix.monitor_gc 0.1 );
+  bindings_pool := make_bindings_pool config.nb_entries;
+  absent_bindings := make_bindings_pool config.nb_entries
```

Comment:
I think you need to check that the values you add in `absent_binding` are not in `binding_pool`.

Reply:
Actually the probability of getting the same key twice is neglectable, so I thought it's not worth the effort of checking since this has a significant cost.

Reply:
I think that it's ok if the setup for a benchmark takes a lot of time. I prefer having a smaller array of absent values if I want a fast setup, but ensuring that they are indeed absent. But that's just my personal preference :)

Reply:
Summary of the discussion we had offline:
1. The probability of having a common element in those two arrays is ~10^-49 for 10_000_000 elements and can go up to ~10^-45 for 1_000_000_000 elements so that is indeed neglectable (can provide the calculus but GitHub is not great at formulas).
2. The time needed *with verification* to generate a size 1000 array of absent values when the main pool is of size 10_000_000 (current behavior) is approximately 90 seconds.
3. The time needed *without verification* (this PR) to generate a 10_000_000 array of (unverified) absent values is 12 seconds; that would not be possible with the verification.
4. If we make a mistake by actually having a common element, the impact on the result would also be neglectable given the size of the dataset.

Given those points, it's probably not worth having the verification.

Side note to mitigate point 1: if we were to use a much smaller key size (<10) the probability would rise beyond neglectability, but I don't think we have a use case for keys that small.